nntrainer/graph/network_graph.cpp

```diff
@@ -221,72 +229,76 @@ int NetworkGraph::realizeMultiOutputType(
  * @fixme: the implementation assumes loss layer should always be at the last
  * layer and the there is only one loss, this assumption is not true
  */
-int NetworkGraph::addLossLayer(const std::string &loss_type) {
-  int status = ML_ERROR_NONE;
-  auto const &last_node = LNODE(graph.getSortedNode(graph.size() - 1));
-  auto last_layer_node = getSortedLayerNode(graph.size() - 1);
-
-  if (last_layer_node->requireLabel()) {
-    if (!istrequal(last_layer_node->getType(), loss_type))
-      status = ML_ERROR_INVALID_PARAMETER;
-    return status;
-  }
-
-  if (loss_type.empty()) {
-    return status;
+void NetworkGraph::finalizeLossLayer() {
+  auto loss_layer_node = getSortedLayerNode(graph.size() - 1);
+  std::string loss_type = loss_layer_node->getType();
+  if (graph.size() == 1) {
```

Comment:
Could you explain what this line is checking?

is it checking that graph is only consists of a loss?

Reply:
For the naive cross-entropy layer, the previous layer must be softmax or sigmoid. So, if graph size is 1, then its not possible. However, other losses/layers are possible.
This also provides added security that the check at L245 does not fail.